homeassistant/components/zha/fan.py

```diff
@@ -43,9 +46,26 @@ async def async_setup_entry(
 class ZhaFan(FanEntity, ZHAEntity):
     """Representation of a ZHA fan."""
 
-    _attr_supported_features = FanEntityFeature.SET_SPEED
     _attr_translation_key: str = "fan"
 
+    def __init__(self, entity_data: EntityData, **kwargs: Any) -> None:
+        """Initialize the ZHA fan entity."""
+        super().__init__(entity_data, **kwargs)
+
+        features: FanEntityFeature = FanEntityFeature(0)
+        zha_features: ZHAFanEntityFeature = self.entity_data.entity.supported_features
+
+        if ZHAFanEntityFeature.SET_SPEED in zha_features:
+            features |= FanEntityFeature.SET_SPEED
+        if ZHAFanEntityFeature.OSCILLATE in zha_features:
+            features |= FanEntityFeature.OSCILLATE
```

Comment:
I believe that these feature flags will require that `async_set_direction` and `async_oscillate` be implemented, which would require parallel methods in the ZHA library: https://developers.home-assistant.io/docs/core/entity/fan/#oscillate

Reply:
Should I just remove them from here? As far as I can tell there is currently no Zigbee Fan that supports those features, however zha references to them. They are listed in https://github.com/zigpy/zha/blob/3a1e0d637a42ed80e323a6d9ecbc406aa25abf52/zha/application/platforms/fan/const.py

Reply:
If there are no fans that use these flags then we should remove them from here. `zigpy/zha` has a lot of unused features that were carried over from Core so many of them will at some point be cleaned up.

Reply:
Done